model/stats/db.go

```diff
@@ -590,6 +600,32 @@ func (filter StatsFilter) buildMatchStageForTest() bson.M {
 		DbTestStatsIdProjectKeyFull:   filter.Project,
 		DbTestStatsIdRequesterKeyFull: bson.M{"$in": filter.Requesters},
 	}
+
+	// After cutting over the below build variants to use resmoke spawned
+	// with jasper (SERVER-54315), test names are now random UUIDs with
+	// with a human-readble display name. For the period between
+	// 09/01/21-09/14/21, the historical test stats calculations failed to
+	// account for this (EVG-15396). We should ignore any test stats
+	// affected by this bug.
+	//
+	// TODO: (EVG-15396) Remove this code once all affected test stats TTL
+	// on 03/14/2022.
+	affectedVariants := []string{
+		"enterprise-rhel-80-64-bit",
+		"enterprise-rhel-80-64-bit-dynamic-required",
+	}
+	if len(filter.BuildVariants) == 0 || len(utility.StringSliceIntersection(filter.BuildVariants, affectedVariants)) > 0 {
+		match["$not"] = []bson.M{
```

Comment:
the [doc](https://docs.mongodb.com/manual/reference/operator/aggregation/not/) just provides a syntax with a single expression in the $not.
Does specifying multiple expressions also work?

Reply:
This is a single expression since it is an array of bson.M of length 1.

Reply:
I don't think the aggregation form of `$not` can be combined with `$match` - I tried it in the mongo shell but it say it's invalid syntax. I think you can only use the non-aggregation form of `$not` in the `$match` query.

Using `$not` in place of `$and` produces an error.
```
db.c.aggregate([{"$match": {"$not": [{"foo": 1}]}}])

uncaught exception: Error: command failed: {
	"operationTime" : Timestamp(1631634685, 1),
	"ok" : 0,
	"errmsg" : "unknown top level operator: $not",
	"code" : 2,
	"codeName" : "BadValue",
	"$clusterTime" : {
		"clusterTime" : Timestamp(1631634685, 1),
		"signature" : {
			"hash" : BinData(0,"AAAAAAAAAAAAAAAAAAAAAAAAAAA="),
			"keyId" : NumberLong(0)
		}
	}
}
```
However, this usage of `$not` is valid:
```
db.c.aggregate([{"$match": {"foo": {"$not": {"$eq": 1}}}])
```

Can we revert the `$not` change and just use the `$and` thing that you used previously?

Reply:
ah, got it. Sorry: I got confused counting brackets 😄

Reply:
Nice catch @Kimchelly. Reverting my suggestion is fine by me.